### PR TITLE
Fix #3374: Add opt-in to Brave Today

### LINKED ISSUE #3290: "Sources & Settings" under intro card sometimes uses grey text
### Description: 

Sometimes the `Sources & Settings` string uses grey text rather than the expected white text. When this happens, it's harder to read as the background of the intro card is grey as well.

### Steps to Reproduce 

I don't really have STR as I've only seen this happen twice. But it usually happens after opening an article from Brave Today. @brave/legacy_qa if you run into this and can reliably reproduce, please update STR 👍 

**Actual result:** 

image

**Expected result:**

image

**Reproduces how often:** [Easily reproduced, Intermittent Issue]

I've only seen this happen a few times so it seems rare but 

**Brave Version:** 

* Reproduced using `1.23 (21.2.10.18)`

**Device details:** 

* `iPhone 11` running `iOS 14.4` using `1.23 (21.2.10.18)` - **Reproduced**
* `iPad Air (3rd Gen)` running `iOS 14.4` using `1.23 (21.2.10.18)` - **Reproduced**

**Website problems only:**
- did you check with Brave Shields down? `N/A`
- did you check in Safari/Firefox (WkWebView-based browsers)? `N/A`

### Additional Information

CCing @karenkliu @anthonypkeane @srirambv @bsclifton @iccub @kylehickinson @soner-yuksel

### LINKED ISSUE #3374: Add opt-in for Brave Today
Drive: `Brave Today Opt-in Spec`

### PR DESCRIPTION
**Pull request by kylehickinson**
## Summary of Changes

This pull request fixes #3374 
This pull request fixes #3290

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:

- See `Brave Today Opt-In Spec` in Drive

## Screenshots:

image

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: BraveShared/BraveStrings.swift [hunk 1]
```diff
@@ -1595,13 +1601,25 @@ extension Strings {
...
             bundle: .braveShared,
-            value: "Your interests are matched on your device so your personal information never leaves your browser. New content updated throughout the day.",
+            value: "Brave Today is ad-supported with completely private and anonymized ads matched on your device.",
+            comment: ""
+        )
+        public static let introCardNew = NSLocalizedString(
+            "today.introCardNew",
+            bundle: .braveShared,
+            value: "New",
+            comment: ""
```
Comment (iccub) [inline BraveShared/BraveStrings.swift]: Add comment what this `new` refers to, many languages use different words for it depending on context.
It should mention something like `new as new feature introduced`

File: Client/Frontend/Brave Today/Cards/BraveTodayOptInView.swift [hunk 2]
```diff
@@ -126,3 +145,52 @@ class BraveTodayWelcomeView: UIView, FeedCardContent {
         }
     }
 }
+
+/// Displays the word "New" with a gradient mask
+private class MaskedNewLabel: UIView {
+    private let gradientView = GradientView(
+//        colors: [UIColor(rgb: 0xA78AFF), UIColor(rgb: 0xFF1893), UIColor(rgb: 0xFA7250)], // dark
```
Comment (iccub) [inline Client/Frontend/Brave Today/Cards/BraveTodayOptInView.swift]: comment

File: Client/Frontend/Settings/Brave Today/BraveTodaySettingsViewController.swift [hunk 3]
```diff
@@ -48,7 +48,16 @@ class BraveTodaySettingsViewController: TableViewController {
         dataSource.sections = [
             .init(
                 rows: [
-                    .boolRow(title: Strings.BraveToday.isEnabledToggleLabel, option: Preferences.BraveToday.isEnabled)
+                    .boolRow(title: Strings.BraveToday.isEnabledToggleLabel, option: Preferences.BraveToday.isEnabled, onValueChange: { value in
```
Comment (iccub) [inline Client/Frontend/Settings/Brave Today/BraveTodaySettingsViewController.swift]: Is `onValueChange` handler safe without weak or unowned `self` here?

**Review by iccub — CHANGES_REQUESTED** [review]
Looks good, add a comment to a NSLocalized string,
general rule I use is the less words are used in a string, the more explanation it needs

you English speakers have it easy, look at this funny example from my language,
all these words refer to word `play`, they differ based on sentence and context

pl_play

**Comment by kylehickinson** [comment]
Haha, good call on the localization comments 😆 Will fix those

File: Client/Frontend/Settings/Brave Today/BraveTodaySettingsViewController.swift (hunk 3 above)
Comment (kylehickinson) [reply Client/Frontend/Settings/Brave Today/BraveTodaySettingsViewController.swift]: Hmm, no I think it'll lead to a retain cycle, will add weak/unowned

**Review by iccub — APPROVED** [review]

**Review by soner-yuksel — APPROVED** [review]

**Pull request merged**